bin/ci-publish.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash -ex
+
+npmPublish() {
+  npm publish
+}
+
+currentVersion() {
+  grep version <package.json | sed 's/[^0-9.]*\([0-9.]*\).*/\1/'
+}
+
+currentProject() {
+  npm ls --depth=-1 | head -1 | cut -f 1 -d " " | sed 's/@.*//'
+}
+
+checkVersion() {
+  local gitTag
+  gitTag=$(git describe --tags)
+
+  if [ "$(currentVersion)" != "$gitTag" ]; then
```

Comment:
Ok, then I think it will work, but only because git has inconsistent behavior, which means this is very deceptive. I think it would be better to make a function called `getTaggedCommitHash` (`git rev-list -n 1 $TAG`) and compare this with the HEAD commit hash; that is a more direct and explicit way of doing it.